pyat/at/physics/energy_loss.py

```diff
@@ -74,6 +74,9 @@ def tracking(ring):
                                      sextupole_pass=None,
                                      octupole_pass=None,
                                      copy=True)
+        diff_elem = ringtmp.get_elements('Diffusion')
+        for de in diff_elem:
+            ringtmp.remove(de)
```

Comment:
`remove` should not be used here:
- its 2nd argument must be an Element, not a list of Elements
- it remove only the 1st Element which is equal to the 2nd argument. There may be several diffusion elements,
- it has to check each element in the lattice for equality with the second argument. It's long, and the quality between Elements is not well defined; it's identity, in fact.
A more efficient way to do that is:
```python
diff_elem = get_cells(ringtmp, checkattr('PassMethod', 'QuantDiffPass')
ringtmp=ringtmp(~diff_elem)
```

That beeing said, I think this has to be done in `Lattice.radiation_off` (called h=just before). Such elements still modify the momentum, so should be turned off in `radiation_off` (set to 'IdentityPass')

Reply:
Moved to `radiaton_off()` as suggested, I also introduced `Element.QuantDiffElement` to simplify detection